cli/azd/pkg/input/paused_console.go

```diff
@@ -0,0 +1,89 @@
+package input
+
+import (
+	"bufio"
+	"context"
+	"io"
+
+	"github.com/azure/azure-dev/cli/azd/pkg/output"
+	"github.com/azure/azure-dev/cli/azd/pkg/output/ux"
+)
+
+// A console implementation which pauses the terminal after each message until the user presses enter
+type pausedConsole struct {
```

Comment:
Naively, I think just adding a `WaitForEnter` method to the Console interface (similar to our other "interactive" methods like Prompt, Select and Confirm) and then implementing it in the few existing implementations we have (AskerConsole and MockConsole are the only two, I think, once we delete MutedConsole) would be a cleaner design.  We'd have less code overall and the intention would be clear in the API itself and at at the call sites vs the current approach where the intention is tied up in policy in this type.

Is there a reason that breaks down in some way?  Or where you using this pattern because we had something similar with MutedConsole?